exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/ParquetPushDownFilter.java

```diff
@@ -165,12 +167,29 @@ protected void doOnMatch(RelOptRuleCall call, FilterPrel filter, ProjectPrel pro
       return;
     }
 
-
     RelNode newScan = ScanPrel.create(scan, scan.getTraitSet(), newGroupScan, scan.getRowType());;
 
     if (project != null) {
       newScan = project.copy(project.getTraitSet(), ImmutableList.of(newScan));
     }
+
+    if (newGroupScan instanceof AbstractParquetGroupScan) {
+      RowsMatch matchAll = RowsMatch.ALL;
+      List<RowGroupInfo> rowGroupInfos = ((AbstractParquetGroupScan) newGroupScan).rowGroupInfos;
+      for (RowGroupInfo rowGroup : rowGroupInfos) {
+        if (rowGroup.getRowsMatch() != RowsMatch.ALL) {
+          matchAll = RowsMatch.SOME;
+          break;
+        }
+      }
+      if (matchAll == ParquetFilterPredicate.RowsMatch.ALL) {
+        call.transformTo(newScan);
+      }
+    } else {
+      final RelNode newFilter = filter.copy(filter.getTraitSet(), ImmutableList.<RelNode>of(newScan));
+      call.transformTo(newFilter);
+    }
+
     final RelNode newFilter = filter.copy(filter.getTraitSet(), ImmutableList.<RelNode>of(newScan));
```

Comment:
How it works in case `newGroupScan` is not an instance of `AbstractParquetGroupScan`? Will not `filter.copy` be called twice?

Reply:
Exact. Suppressed duplicate lines